src/components/Common/SkeletonLoader/SkeletonLoader.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './SkeletonLoader.styl';
+import { Block, Elem } from '../../../utils/bem';
+
+export const SkeletonLoader = () => {
+
+  return (
+    <Block name="skeletonLoader">
+      <Elem name='title'></Elem>
+      <Elem name='mainValue'></Elem>
+      <Elem name='additionalData'></Elem>
```

Comment:
I like it, but was wondering if we could make this a bit more reusable as skeleton loaders can apply to a wide variety of scenarios. Also there are already styles to achieve skeleton loading somewhere else in the codebases. I'm just trying to ensure we can culminate and collapse that as we shift towards a more cohesive repo set. To keep that focus, I think promoting a clear way to maintain these loaders makes a whole lot of sense.

Reply:
You know, maybe removing the internal line items directly and having each sit as a specific component that can facilitate shapes instead.

And using those create configurable loaders that can serve a variety of use cases.

ex. SkeletonLine (line count = 1), SkeletonAvatar (shape = "circle" | "square")

Then you can create nearest the usages:
```tsx
const TaskCardSkeleton = () => (
   <SkeletonLoader>
     <SkeletonLine />
     <SkeletonLine lineCount={2} />
     <SkeletonLine lineCount={4} />
   </SkeletonLoader>
)

const ProjectCardSkeleton = () => (
   <SkeletonLoader>
     <SkeletonLine />
     <SkeletonLine lineCount={4} />
     <Flex gap="8px">
       <Flex>
         <SkeletonAvatar shape="circle" />
         <SkeletonAvatar shape="circle" />
         <SkeletonAvatar shape="circle" />
       </Flex>
       <SkeletonLine />
     </Flex>
   </SkeletonLoader>
)
```

And so on. This way we have the basis and building blocks to achieve any form we need started, and implement the current use case using it. We can then formally make this the standard as we transition towards a more holistic design system.

Reply:
agreed that makes it way more usable! and we need to get into the habbit of reusing these kinds of components where possible

Reply:
ok I got that working with some safe defaults that would match existing defaults - please take a peak and let me know your thoughts. im holding off on Flex and SkeletonAvatar for now since we dont really have a working example of where they would be used (Flex could likely be an unrelated component to be added later as a "Structural" component that allows us to help position and line things up with out having to constantly add display flex styling every where)